### PR TITLE
feat: CLIN-2357 ajout colonne code du ministere

### PR DESCRIPTION
**Pull request by claudia1296**
# FEAT:  Ajouter colonne ‘Code du ministère’ 

- closes #CLIN-2357

## Description
Ajouter colonne ‘Code du ministère’ mettre en 2eme position du tableau

[[JIRA LINK]](https://ferlab-crsj.atlassian.net/browse/CLIN-2357)

### After
image

## QA

Steps to validate
Url (storybook, ...)
...

### CODE REVIEW

File: src/graphql/prescriptions/queries.tsx [hunk 1]
```diff
@@ -147,10 +147,9 @@ export const ANALYSIS_ENTITY_QUERY = (requestId: string) => gql`
         text
       }
       code @flatten {
-        coding(system: "http://fhir.cqgc.ferlab.bio/CodeSystem/analysis-request-code")
-          @flatten
-          @first {
+        coding{
```
Comment (creativeyann17) [inline src/graphql/prescriptions/queries.tsx:150]: Revert that change (also it had side-effects making prescription.code string -> array making `src/views/Prescriptions/Detail/AnalysisCard/index.tsx:48` silently fails, once revert will be fine)

Easy change to retreive the data you want is in that file but different place:

**ANALYSIS_PATIENT_FRAGMENT**

Update the **requests** part as bellow;

```
 requests: ServiceRequestList(_reference: patient, based_on: "${requestId}") {
      id
      authoredOn
      code @flatten {
        coding{
          code
          system
        }
      }
...
```

File: src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx [hunk 2]
```diff
@@ -21,12 +19,30 @@ import { Paraclinique } from './components/Paraclinique';
...
 import styles from './index.module.scss';
 
+const { Title } = Typography;
 type OwnProps = {
   prescription?: ServiceRequestEntity;
   loading: boolean;
 };
 
 const ClinicalInformation = ({ prescription, loading }: OwnProps) => {
+  const [resquestList, setRequestList] = useState<ServiceRequestEntity[]>([]);
```
Comment (creativeyann17) [inline src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx:29]: can be revert, your data is now available in `prescription?.subject?.resource?.requests[].coding[]`

File: src/api/fhir/index.ts [hunk 3]
```diff
@@ -17,6 +19,18 @@ const searchPractitionerRole = () =>
     },
   });
 
+export const fetchServiceRequestEntity = (id: string) =>
```
Comment (creativeyann17) [inline src/api/fhir/index.ts:22]: No need for a new request, see bellow

File: src/views/Prescriptions/Detail/ParentCard/index.tsx [hunk 4]
```diff
@@ -19,46 +21,65 @@ interface OwnProps {
...
-        </Space>
-      </CollapsePanel>
-    )}
-  </ParagraphLoader>
-);
+const ParentCard = ({ extension, loading }: OwnProps) => {
+  const [resquestList, setRequestList] = useState<ServiceRequestEntity[]>([]);
+  const [requestLoading, setRequestLoading] = useState<boolean>(true);
+  useEffect(() => {
+    const subjectRequest = extension?.extension?.[1].valueReference?.resource.requests;
```
Comment (creativeyann17) [inline src/views/Prescriptions/Detail/ParentCard/index.tsx:28]: same here no new query needed and the data should be available in `prescription?.subject?.resource?.requests[].coding[];`

**Review by creativeyann17 — CHANGES_REQUESTED** [review]
:cookie:

**Review by creativeyann17 — APPROVED** [review]

File: src/views/Prescriptions/Detail/RequestTable.tsx [hunk 5]
```diff
@@ -24,6 +24,15 @@ const getRequestColumns = (): TableColumnType<Record<string, any>>[] => [
     title: intl.get('screen.prescription.entity.request.id'),
     render: (id) => extractServiceRequestId(id),
   },
+  {
+    key: 'coding',
+    dataIndex: 'coding',
+    title: intl.get('screen.prescription.entity.request.code'),
+    render: (coding: { code: string; system: string }[]) => {
+      const sequencingCode = coding?.find((c) => c.system?.includes('sequencing'));
+      return sequencingCode?.code;
```
Comment (atoulous) [inline src/views/Prescriptions/Detail/RequestTable.tsx:33]: `sequencingCode?.code || TABLE_EMPTY_PLACE_HOLDER`
vous faites pas ca en général ?

File: src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx [hunk 6]
```diff
@@ -118,7 +115,7 @@ const ClinicalInformation = ({ prescription, loading }: OwnProps) => {
           </div>
           <RequestTable
             patientId={extractPatientId(prescription?.subject.reference)}
-            data={prescription?.subject?.resource?.requests}
+            data={prescription.subject?.resource?.requests}
```
Comment (atoulous) [inline src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx:118]: pourquoi avoir retirer le `?` 
ca sécurise

File: src/graphql/prescriptions/queries.tsx [hunk 7]
```diff
@@ -82,6 +82,7 @@ const ANALYSIS_PATIENT_FRAGMENT = (requestId: string) => gql`
     requests: ServiceRequestList(_reference: patient, based_on: "${requestId}") {
       id
       authoredOn
+      code @flatten { coding{ code system } }
```
Comment (atoulous) [inline src/graphql/prescriptions/queries.tsx:85]: `coding {`
il y a des extensions d'IDE pour graphql qui permettent le lintage

Comment (claudia1296) [reply src/graphql/prescriptions/queries.tsx]: :+1:

File: src/views/Prescriptions/Detail/RequestTable.tsx (hunk 5 above)
Comment (claudia1296) [reply src/views/Prescriptions/Detail/RequestTable.tsx]: bon point j'avais oublier cette partie

File: src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx (hunk 6 above)
Comment (atoulous) [reply src/views/Prescriptions/Detail/ClinicalInformationCard/index.tsx]: ah exact

**Review by atoulous — APPROVED** [review]

**Pull request merged**